### PR TITLE
Add hbi.api.disable-xjoin feature flag

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-xxxx](https://issues.redhat.com/browse/RHINENG-xxxx).

* Add feature flag and fall back
* Add a custom strategy so the feature can be applied only to select orgs for testing

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -230,7 +231,11 @@ def create_app(runtime_environment):
         flask_app.config["UNLEASH_APP_NAME"] = "host-inventory-api"
         flask_app.config["UNLEASH_ENVIRONMENT"] = "default"
         flask_app.config["UNLEASH_URL"] = app_config.unleash_url
-        flask_app.config["UNLEASH_CUSTOM_HEADERS"] = {"Authorization": f"Bearer {app_config.unleash_token}"}
+        if app_config.is_clowder:
+            flask_app.config["UNLEASH_CUSTOM_HEADERS"] = {"Authorization": f"Bearer {app_config.unleash_token}"}
+        else:
+            flask_app.config["UNLEASH_CUSTOM_HEADERS"] = {"Authorization": f"{app_config.unleash_token}"}
```
Comment (kruai) [inline app/__init__.py:237]: Could we just prepend "Bearer" to the `unleash_token` env var so we don't need the clowder check?

File: README.md [hunk 2]
```diff
@@ -174,6 +174,9 @@ honcho start
  INVENTORY_LOGGING_CONFIG_FILE=logconf.ini
  INVENTORY_DB_SSL_MODE=""
  INVENTORY_DB_SSL_CERT=""
+ UNLEASH_TOKEN='*:*.dbffffc83b1f92eeaf133a7eb878d4c58231acc159b5e1478ce53cfc'
```
Comment (kruai) [inline README.md:177]: Where is this value from? I'm a bit nervous as looks like a secret

**Review by kruai — COMMENTED** [review]
Thanks for this - looks good, just have a couple Qs

File: app/__init__.py (hunk 1 above)
Comment (chambridge) [reply app/__init__.py:237]: Its a Bearer token when clowder is in use and an API key in this local container flow

File: README.md (hunk 2 above)
Comment (chambridge) [reply README.md:177]: Not a secret just a dummy value that is used to prep the container here:
https://github.com/RedHatInsights/insights-host-inventory/pull/1593/files#diff-26fac0c0264ace68fab9c60b467544b6c45f3f46d82c270a73bc28681fd45096R37

Just reused the same value from another project: https://github.com/project-koku/koku/blob/main/.env.example#L27
But if you read the documentation here: https://docs.getunleash.io/using-unleash/deploy/configuring-unleash

> You can also use the environment variables INIT_ADMIN_API_TOKENS, INIT_CLIENT_API_TOKENS, and INIT_FRONTEND_API_TOKENS. All variables require a comma-separated list of API tokens to initialize (for instance *:*.some-random-string, *:*.some-other-token). The tokens found in INIT_ADMIN_API_TOKENS, INIT_CLIENT_API_TOKENS, and INIT_FRONTEND_API_TOKENS will be created as admin, client, and frontend tokens respectively, and Unleash will assign usernames automatically.

**Comment by thearifismail** [comment]
@chambridge how is this PR connected to switching to DB and removing xjoin?

**Comment by chambridge** [comment]
> @chambridge how is this PR connected to switching to DB and removing xjoin?

Introduces the feature flag that will be used to control (on a per org basis) whether we use xjoin or direct db access until we are ready to switch completely to direct db access.

In this follow on work you can see how the feature flag is used to enable this: https://github.com/chambridge/insights-host-inventory/commit/4bf0d3ef50f486b32043ee7d1e99380cd962f79a

File: app/__init__.py (hunk 1 above)
Comment (kruai) [reply app/__init__.py:237]: Oh - guess this is necessary, then

**Review by kruai — APPROVED** [review]

**Pull request merged**